src/frules/basic.jl

```diff
@@ -0,0 +1,11 @@
+frule!!(f::F, args::Vararg{Dual,N}) where {F,N} = frule!!(zero_dual(f), args...)
```

Comment:
yes but I'm wondering if it's a bad thing to leave it there?

Reply:
I would much rather aim for complete uniformity here -- it's what we do in reverse-mode, and I think it will ensure that we avoid and "perturbation confusion" problems when doing forwards-mode over forwards-mode (if you know that the second time you differentiate a programme, everything is going to be a `Dual` inside a `Dual`, you avoid issues of the form "okay, so this is a `Dual`, but is it a `Dual` associated to the first time I differentiated this programme, or the second? Off the top of my head, I don't have a concrete example to put forward, but we should definitely try and concoct one -- there might be one somewhere in ForwardDiff.jl, but if not there must be plenty of examples out there in the literature)

Reply:
See https://github.com/compintell/Mooncake.jl/pull/389#issuecomment-2499826050, I replaced this with `_frule!!_funcnotdual` to avoid confusion